File: hw5/hw5.FS/Calculator.fs
module hw5.FS.Calculator
open hw5.ResBuilder
let tryDivide arg1 arg2 =
    if arg2 = decimal 0 then Error "Second argument is zero!"
    else Ok(arg1 / arg2)
let Calculate (val1, operation, val2) =
    res{match operation with
        | Operation.Plus -> return val1 + val2
        | Operation.Minus -> return val1 - val2
        | Operation.Multiply -> return val1 * val2
        | _ -> let! res =  tryDivide val1 val2
               return res}

File: hw12/hw12/AspProj/Services/Calculator.cs
using System;

namespace hw8.Services
{
    public class Calculator: ICalculator
    {
        public double Add(double arg1, double arg2) => arg1 + arg2;

        public double Subtract(double arg1, double arg2) => arg1 - arg2;

        public double Divide(double arg1, double arg2) => arg2 == 0 ?
            double.NaN : arg1 / arg2;

        public double Multiply(double arg1, double arg2) => arg1 * arg2;
    }
}